src/octree/octree_iterator.rs

```diff
@@ -66,32 +66,30 @@ fn get_node_iterator(octree: &Octree, node_id: &NodeId) -> NodeIterator<RawNodeR
 }
 
 /// iterator over the points of a octree node that satisfy the condition expressed by a boolean function
-pub struct FilteredPointsIterator<F> {
-    filter_func: F,
+pub struct FilteredPointsIterator {
+    culling: Box<dyn PointCulling<f64>>,
```

Comment:
this means that the filtering now is restricted only by culling/geolocation and not by any other function (idk..eg all points white, all points with intensity > x..)

Reply:
For general filtering, there is `filter()`, so I don't think we need to have that here. This change is needed because you cannot name the type of a `FilteredPointsIterator<F>` when writing the instance of `PointCloud`.